### PR TITLE
added ngenius payment link processing on webhook

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Payments/External/PaymentLinks/IPaymentLinksProcessingService.cs [hunk 1]
```diff
@@ -13,6 +14,8 @@ public interface IPaymentLinksProcessingService
         Task<Result<PaymentResponse>> Pay(string code, NGeniusPayByLinkRequest request, string ip, string languageCode);
 
         Task<Result<PaymentResponse>> ProcessResponse(string code, JObject value);
+        
+        Task<Result<PaymentResponse>> ProcessResponse(string code, CreditCardPaymentStatuses status);
```
Comment (mingazhev) [inline Api/Services/Payments/External/PaymentLinks/IPaymentLinksProcessingService.cs:18]: May we differ these two methods? Responses are different for payment processors

File: Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs [hunk 2]
```diff
@@ -71,6 +69,19 @@ public Task<Result<PaymentResponse>> ProcessResponse(string code, JObject respon
...
 
 
+        public Task<Result<PaymentResponse>> ProcessResponse(string code, CreditCardPaymentStatuses status)
+        {
+            return Result.Success()
+                .BindWithLock(_locker, typeof(PaymentLink), code, () => Result.Success()
+                    .Bind(GetLink)
+                    .Bind(ProcessResponse));
+
+            Task<Result<PaymentLink>> GetLink() => this.GetLink(code);
```
Comment (mingazhev) [inline Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs:79]: Arrow to the next line

File: Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs [hunk 3]
```diff
@@ -194,6 +205,42 @@ async Task<PaymentResponse> StorePaymentResult(PaymentResponse paymentResponse)
...
+                    string.Empty));
+            }
+
+
+            bool ShouldSendReceipt(PaymentResponse parsedResponse)
+            {
+                return status == CreditCardPaymentStatuses.Success &&
+                    IsNotAlreadyPaid(link);
+
+                static bool IsNotAlreadyPaid(PaymentLinkData link) => link.CreditCardPaymentStatus != CreditCardPaymentStatuses.Success;
```
Comment (mingazhev) [inline Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs:230]: arrow to the next line everywhere

File: Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs]: Fixed

File: Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Payments/External/PaymentLinks/PaymentLinksProcessingService.cs]: Fixed

File: Api/Services/Payments/External/PaymentLinks/IPaymentLinksProcessingService.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Payments/External/PaymentLinks/IPaymentLinksProcessingService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**